src/common/pecan.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef PECAN_H
+#define PECAN_H
+#include "./ptypes.hpp"
+
+#define Param_Count 20
+#define MAX_SIZE_PACKET_DATA 8
+
+// Various PCAN Error Code return values
+enum PCAN_ERR{
+    PACKET_TOO_BIG=-3,
+    NOT_RECEIVED=-2,
+    NOSPACE=-1,
+    SUCCESS = 0,
+    GEN_FAILURE = 1,
+};
+
+// Exact match or Similar match (with mask)
+enum MATCH_TYPE{
+    MATCH_EXACT=0,
+    MATCH_SIMILAR=1,
+};
+
+// A CANPacket: takes in 11-bit id, 8 bytes of data
+struct CANPacket {
+    int id;
+    char data[MAX_SIZE_PACKET_DATA] = {0};
+    char dataSize = 0;
```

Comment:
Perhaps `uint8_t` might be a better choice since it's more explicit and not platform dependent.

Reply:
but thank you for your feedback!!!